xarray/testing/strategies.py

```diff
@@ -0,0 +1,452 @@
+from collections.abc import Hashable, Iterable, Mapping, Sequence
+from typing import TYPE_CHECKING, Any, Protocol, Union, overload
+
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+import numpy as np
+from hypothesis.errors import InvalidArgument
+
+import xarray as xr
+from xarray.core.types import T_DuckArray
+
+if TYPE_CHECKING:
+    from xarray.core.types import _DTypeLikeNested, _ShapeLike
+
+
+__all__ = [
+    "supported_dtypes",
+    "names",
+    "dimension_names",
+    "dimension_sizes",
+    "attrs",
+    "variables",
+    "unique_subset_of",
+]
+
+
+class ArrayStrategyFn(Protocol[T_DuckArray]):
+    def __call__(
+        self,
+        *,
+        shape: "_ShapeLike",
+        dtype: "_DTypeLikeNested",
+    ) -> st.SearchStrategy[T_DuckArray]:
+        ...
+
+
+def supported_dtypes() -> st.SearchStrategy[np.dtype]:
+    """
+    Generates only those numpy dtypes which xarray can handle.
+
+    Avoiding using hypothesis.extra.numpy.scalar_dtypes is required to exclude weirder dtypes e.g. unicode, byte_string, array, or nested dtypes.
+    Also required to dodge bugs with pandas non-nanosecond datetime overflows.
+    Note does not generate all dtypes that xarray can handle - just only generates dtypes which it definitely can.
+
+    Requires the hypothesis package to be installed.
```

Comment:
I wonder if we even need these caveats? You can only import the package if `hypothesis` is installed, and "search strategy" does not really make sense otherwise. However, I would add "see also" links that point to the user guide.

Reply:
I've added "See Also" links to all strategies.